rlclientlib/logger/event_logger.cc

```diff
@@ -2,28 +2,72 @@
 #include "ranking_event.h"
 #include "err_constants.h"
 #include "time_helper.h"
+
+#include <functional>
+#include <memory>
 namespace reinforcement_learning { namespace logger {
   int interaction_logger::log(const char* event_id, string_view context, unsigned int flags, const ranking_response& response, api_status* status, learning_mode learning_mode) {
     const auto now = _time_provider != nullptr ? _time_provider->gmt_now() : timestamp();
-    return append(ranking_event::choose_rank(event_id, context, flags, response, now, 1.0f, learning_mode), status);
+    // using shared_ptr because we can't move a unique_ptr in C++11
+    // We should replace them in C++14
+    auto evt_sp = std::make_shared<ranking_event>();
```

Comment:
Switch to raw pointer here.  The usage and life time is very clear.  The overhead of shared_ptr seems unnecessary given we are doing this for perf reasons.

Reply:
I think we should avoid switching to a raw pointer at all costs. Shared pointers mostly cause a perf hit if they are actually shared this one here seems to just being moved around.

If there is a measurable perf hit then we can consider it

Reply:
+1 to Olga's comment. Safety should be the default approach unless there is strong evidence of a performance issue.

Reply:
discussed outside the PR: A smart object of some kind is required because the functor will never get run for dropped events (eg: when being pruned) which would mean the objects never get destroyed. Rajan to investigate alternatives when convenient